### PR TITLE
Modify inertia implementation for freelook and orbit

### PR DESCRIPTION
**Pull request by toger5**
This is a solution for the topics discussed in: #10994 #10667

#10994 gets superseded by this one too since I needed the settings to do effective testing.
@kubecz3k I hope you are okay with that. otherwise we could ask @akien-mga to merge your pr first and than I rebase and adapt it.

The only real issue that gets fixed though is:
 - inertia calculation was frame rate dependant. (now it feels about the same)
 - The comment @reduz mention in this commit message 8951403 (this inertia implementation works in orthogonal too)

## Changes:
 - freelook now uses position based inertia (holding a key for a
   specific time always rults in the same distance traveled independent
of inertia setting)
 - orbit inertia now is angle bases. (not transformation based) ->
   camera always takes the same path.
 - added setting for orbit inertia
 - added setting hints for freelook settings.

### orbit inertia now is angle bases. (not transformation based) -> camera always takes the same path.

In this video (https://youtu.be/e85LIZTGEns) it is really good to see how the old transform dependant orbit interpolation affected the camera path. the camera gets closer to the rotation centre with high inertia and fast inputs.
Now the path will always be the same, just the movement will be smooth

this image illustrates how the inertia did affect ("current Movement") the camera path. and how the movement will be with this pr ("Movement when the angle is interpolated")
screen shot 2017-09-06 at 16 06 42

but still If you refer the old behaviour (like shown in the video) than we need another pr I guess.

### CODE REVIEW

**Comment by groud** [comment]
Does this might fix the comment here : https://github.com/godotengine/godot/commit/895140389a7306eafc9a36f51d892ece0001f223 ?

**Comment by toger5** [comment]
@groud that is a good idea. it might actually work since I don't anymore interpolate the transformation of the camera, instead I interpolate the angles stored in the Cursor object.

**Comment by groud** [comment]
Ok, maybe you could check if it still works without @reduz's change ? :)

**Comment by toger5** [comment]
@groud yea it definitely has inertia for orthogonal camera with this branch.

**Comment by toger5** [comment]
Oh reduz committed that just right now... I was wondering why I there was now `orthogonal ||` in the first place.

**Comment by groud** [comment]
Ok then maybe you could rebase and revert reduz changes then. :) 

> Oh reduz committed that just right now... I was wondering why I there was now orthogonal || in the first place.

Yeah that's why ^^

**Comment by toger5** [comment]
should I revert red commit in another commit or should I just override his change in the process of rebasing?

**Comment by groud** [comment]
As you want, I think you can add it in your own commit (with a `git commit --amend`), after a rebase. :)

**Comment by kubecz3k** [comment]
@toger5 sure I'm fine. Is it ok to close https://github.com/godotengine/godot/pull/10994 then ?

**Comment by toger5** [comment]
@groud too late I added the revert... so it's two commits now. (but I think this is better for the history since it is now obvious, that the change gets overridden)

**Comment by toger5** [comment]
@kubecz3k maybe you could try if you like how the settings behave (I guess you did better testing than me...) and if you are okay you can close #10994. (or you wait until it is merged, test, and do another pr to optimise the settings)

**Comment by groud** [comment]
> @groud too late I added the revert... so it's two commits now. (but I think this is better for the history since it is now obvious, that the change gets overridden)

Yeah no problem :)

File: editor/plugins/spatial_editor_plugin.cpp [hunk 1]
```diff
@@ -1689,60 +1685,49 @@ void SpatialEditorViewport::_update_freelook(real_t delta) {
...
 	// Higher inertia should increase "lag" (lerp with factor between 0 and 1)
-	// Inertia of zero should produce instant movement (lerp with factor of 1)
-	// Takes reference of 60fps for units, so that inertia of 1 gives approximate lerp factor of 0.5
-	real_t factor = 1.0 / (1.0 + inertia * delta * 60.f);
-	freelook_velocity = freelook_velocity.linear_interpolate(instant_velocity, CLAMP(factor, 0, 1));
+	// Inertia of zero should produce instant movement (lerp with factor of 1) in this case it returns a really high value and gets clamped to 1.
 
-	cursor.pos += freelook_velocity * delta;
+	freelook_target_position += direction * speed;
+	real_t factor = (1.0 / (inertia + 0.001)) * delta;
```
Comment (toger5) [inline editor/plugins/spatial_editor_plugin.cpp]: @zylann this is the factor calculation I used. (it is the one you proposed in the code snipped)

**Review by toger5 — COMMENTED** [review]
Just a hint for @Zylann if you are okay with that implementation.

File: editor/plugins/spatial_editor_plugin.cpp (hunk 1 above)
Comment (Zylann) [reply editor/plugins/spatial_editor_plugin.cpp]: Seems fine

**Comment by reduz** [comment]
I did some changes to this, so zoom and translation can also be interpolated.
Also made sure there is a distinction between camera being manipulated and not (like when pressing F, changing view, etc), so motion is softer in later cases.. this allows transitioning from one place to the other without losing track of where you are (which was lost in this PR).

Please ask me if you plan to merge further changes, so we avoid stepping on each other toes.

**Comment by toger5** [comment]
@reduz I also did a PR for that: #11222
Should have communicated earlier, that * did some more changes. 
Ill Check your commit probably the Implementation is similar. So it does not really matter who does it.

**Pull request merged**